pkg/kubectl/resource/mapper.go

```diff
@@ -38,7 +38,7 @@ type Mapper struct {
 // InfoForData creates an Info object for the given data. An error is returned
 // if any of the decoding or client lookup steps fail. Name and namespace will be
 // set into Info if the mapping's MetadataAccessor can retrieve them.
-func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
+func (m *Mapper) InfoForData(data []byte, source string, local bool) (*Info, error) {
```

Comment:
Instead of plumbing `local` all the way through the layers, couldn't you pass a different `clientMapper` to `NewBuilder`?  If I'm reading the rest of the code correctly, its currently plumbing the value through all the layers to adjust the `clientMapper` to always return `nil`.